syntax = "proto3";

package tcash.evm_engine.parameters;

import "proto_basic.proto";

message SubmitResult {
    uint32 version = 1;
    uint32 transaction_status = 2;
    bytes status_data = 3;
    uint64 gas_used = 4;
    repeated basic.ResultLog logs = 5;
}

message FunctionCallArgs {
    uint32 version = 1;
    basic.ProtoAddress address = 2;
    basic.WeiU256 value = 3;
    bytes input = 4;
    uint64 gas_limit = 5;
}